### PR TITLE
[IOS-567, IOS-568] Fix Ethereum address validation and input field resizing

### PR DESCRIPTION
**Pull request by yuliaveres**
Address field ***not resized*** on Send token view + address ***validation*** does not work properly as not the whole string is taken into consideration:
img_0123

***Fixed:***
img_0124
img_0125

### CODE REVIEW

File: Toshi/Models/EthereumAddress.swift [hunk 1]
```diff
@@ -40,7 +40,7 @@ struct EthereumAddress {
 
     static func validate(_ address: String) -> Bool {
         do {
-            let regex = try NSRegularExpression(pattern: "0x[a-fA-F0-9]{40}")
+            let regex = try NSRegularExpression(pattern: "^0x[a-fA-F0-9]{40}$")
```
Comment (designatednerd) [inline Toshi/Models/EthereumAddress.swift]: Out of curiosity: What do these changes to the Regex do to make this work better?

File: Tests/EthereumAddressTests.swift [hunk 2]
```diff
@@ -35,4 +35,17 @@ class EthereumAddressTests: XCTestCase {
...
+        XCTAssertTrue(EthereumAddress.validate(canonicalAddress))
+
+        let tooLongAddress = canonicalAddress + "j89"
+        XCTAssertFalse(EthereumAddress.validate(tooLongAddress))
+
+        let tooShortAddress = String(canonicalAddress.prefix(10))
+        XCTAssertFalse(EthereumAddress.validate(tooShortAddress))
+
+        let nonHexAddress = String(canonicalAddress.suffix(canonicalAddress.count - 3))
+        XCTAssertFalse(EthereumAddress.validate(nonHexAddress))
```
Comment (designatednerd) [inline Tests/EthereumAddressTests.swift]: are there examples of valid addresses which were previously failing which we should be testing here?

File: Toshi/Models/EthereumAddress.swift (hunk 1 above)
Comment (yuliaveres) [reply Toshi/Models/EthereumAddress.swift]: so the regex checks for the whole string, not just finds valid substring and all good

File: Tests/EthereumAddressTests.swift (hunk 2 above)
Comment (yuliaveres) [reply Tests/EthereumAddressTests.swift]: no, invalid addresses were succeeding

**Pull request merged**